lib/dao.js

```diff
@@ -2407,6 +2577,190 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+DataAccessObject.prototype.replaceAttributes = function(data, options, cb) {
+  var Model = this.constructor;
+  var id = getIdValue(this.constructor, this);
+  var inst = this;
+  var strict = this.__strict;
+  return Model.replaceById(id, inst, Model, data, options, strict, cb);
+};
+
+DataAccessObject.replaceById = function(id, inst, Model, data, options, strict, cb) {
```

Comment:
@Amir-61 Yes, `replaceById` should be `function(id, data, options, cb)`. 

The trick is to build build a new instance using the provided `data` - take a look at how `DataAccessObject.create` is implemented for example: https://github.com/strongloop/loopback-datasource-juggler/blob/63482ef0619bfff429e807bfa2af6ea8b6c00edc/lib/dao.js#L277-L297

```js
// trimmed-down version to illustrate my point, see DAO.create for the full version
DataAccessObject.replaceById = function(id, data, options, cb) {
  var Model = this;
  var inst = new Model(data);
  // etc.
};
```